File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.java
import java.util.*;
class Solution {
    public int[] solution(int l, int r) {
        ArrayList<Integer> answer = new ArrayList<>();
        for (int i = l; i <= r; i++) {
            String numStr = Integer.toString(i);
            boolean valid = true;
            for (int j = 0; j < numStr.length(); j++) {
                char ch = numStr.charAt(j);
                if (ch != '0' && ch != '5') {
                    valid = false;
                    break;
                }
            }
            if (valid) {
                answer.add(i);
            }
        }
        
        if (answer.isEmpty()) {
            return new int[] {-1};
        }
        
        int[] result = new int[answer.size()];
        for (int i = 0; i < answer.size(); i++) {
            result[i] = answer.get(i);
        }
        return result;
    }
}

File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.js
function solution(l, r) {
    let answer = [];
    for(let i = l; i<=r; i++){
        let str = i.toString();
        if([...str].every(ch => ch ==='5' || ch ==='0'))
            answer.push(i);
        
    }
    if(answer.length == 0) answer.push(-1);
    return answer;
}